app/services/katello/ui_notifications/pulp/proxy_disk_space.rb

```diff
@@ -0,0 +1,39 @@
+module Katello
+  module UINotifications
+    module Pulp
+      class ProxyDiskSpace
+        class << self
+          def deliver!
+            SmartProxy.unscoped.with_features('Pulp').each do |proxy|
+              percentage = proxy.statuses[:pulp].storage['pulp_dir']['percent']
+              next unless percentage[0..2].to_i > 90
+              next if notification_already_exists?(proxy)
+              ::Notification.create!(
+                :subject => proxy,
+                :initiator => User.anonymous_admin,
+                :audience => Notification::AUDIENCE_ADMIN,
+                :message => ::UINotifications::StringParser.new(
+                  blueprint.message,
+                  :subject => proxy,
+                  :percentage => percentage
+                ),
+                :notification_blueprint => blueprint
+              )
+            end
+          end
+
+          def notification_already_exists?(subject)
```

Comment:
when do you expire the notification? eg if the disk space state is "resolved" I would suggest to either create a new notification, or remove the existing one?